src/autopas/containers/cellPairTraversals/SlicedTraversal.h

```diff
@@ -164,4 +174,65 @@ inline void SlicedTraversal<ParticleCell, PairwiseFunctor, useSoA, useNewton3>::
   }
 }
 
-}  // namespace autopas
\ No newline at end of file
+template <class ParticleCell, class PairwiseFunctor, bool useSoA, bool useNewton3>
+template <class Particle>
+inline void SlicedTraversal<ParticleCell, PairwiseFunctor, useSoA, useNewton3>::traverseCellVerlet(
+    std::vector<std::vector<std::pair<Particle *, std::vector<Particle *>>>> &verlet) {
+  using std::array;
+
+  auto numSlices = _sliceThickness.size();
+  // 0) check if applicable
+
+  for (size_t i = 0; i < numSlices - 1; ++i) {
+    locks[i] = new autopas_lock_t;
+    autopas_init_lock(locks[i]);
+  }
+
+#ifdef AUTOPAS_OPENMP
+// although every thread gets exactly one iteration (=slice) this is faster than
+// a normal parallel region
+#pragma omp parallel for schedule(static, 1)
+#endif
+  for (size_t slice = 0; slice < numSlices; ++slice) {
+    array<unsigned long, 3> myStartArray{0, 0, 0};
+    for (size_t i = 0; i < slice; ++i) {
+      myStartArray[_dimsPerLength[0]] += _sliceThickness[i];
+    }
+
+    // all but the first slice need to lock their starting layer.
+    if (slice > 0) {
+      autopas_set_lock(locks[slice - 1]);
+    }
+    for (unsigned long dimSlice = myStartArray[_dimsPerLength[0]];
+         dimSlice < myStartArray[_dimsPerLength[0]] + _sliceThickness[slice]; ++dimSlice) {
+      // at the last layer request lock for the starting layer of the next
+      // slice. Does not apply for the last slice.
+      if (slice != numSlices - 1 && dimSlice == myStartArray[_dimsPerLength[0]] + _sliceThickness[slice] - 1) {
+        autopas_set_lock(locks[slice]);
+      }
+      for (unsigned long dimMedium = 0; dimMedium < this->_cellsPerDimension[_dimsPerLength[1]] - 1; ++dimMedium) {
+        for (unsigned long dimShort = 0; dimShort < this->_cellsPerDimension[_dimsPerLength[2]] - 1; ++dimShort) {
+          array<unsigned long, 3> idArray;
+          idArray[_dimsPerLength[0]] = dimSlice;
+          idArray[_dimsPerLength[1]] = dimMedium;
+          idArray[_dimsPerLength[2]] = dimShort;
+          auto baseIndex = utils::ThreeDimensionalMapping::threeToOneD(idArray, this->_cellsPerDimension);
+          this->iterateVerletListsCell(verlet, baseIndex);
+        }
+      }
+      // at the end of the first layer release the lock
+      if (slice > 0 && dimSlice == myStartArray[_dimsPerLength[0]]) {
+        autopas_unset_lock(locks[slice - 1]);
+      } else if (slice != numSlices - 1 && dimSlice == myStartArray[_dimsPerLength[0]] + _sliceThickness[slice] - 1) {
```

Comment:
document what this `else if` statement is used for?
(clearing of the lock set on the last layer of each slice.)